storage/store/chunk_data_packs_test.go

```diff
@@ -20,113 +20,258 @@ import (
+		unittest.RunWithPebbleDB(t, func(chunkDataPackPdb *pebble.DB) {
+			lockManager := storage.NewTestingLockManager()
+			protocolDB := pebbleimpl.ToDB(protocolPdb)
+			chunkDataPackDB := pebbleimpl.ToDB(chunkDataPackPdb)
 
-		// Removing again should not error
-		require.NoError(t, chunkDataPackStore.Remove(chunkIDs))
+			transactions := store.NewTransactions(&metrics.NoopCollector{}, protocolDB)
+			collections := store.NewCollections(protocolDB, transactions)
+			stored := store.NewStoredChunkDataPacks(&metrics.NoopCollector{}, chunkDataPackDB, 10)
+			chunkDataPackStore := store.NewChunkDataPacks(&metrics.NoopCollector{}, protocolDB, stored, collections, 1)
+
+			chunkDataPacks := unittest.ChunkDataPacksFixture(5)
+			for _, chunkDataPack := range chunkDataPacks {
+				// store collection in Collections storage
+				_, err := collections.Store(chunkDataPack.Collection)
+				require.NoError(t, err)
+			}
+
+			// Store chunk data packs
+			require.NoError(t, unittest.WithLock(t, lockManager, storage.LockInsertOwnReceipt, func(lctx lockctx.Context) error {
+				storeFunc, err := chunkDataPackStore.Store(chunkDataPacks)
+				if err != nil {
+					return err
+				}
+				return protocolDB.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
+					return storeFunc(lctx, rw)
+				})
+			}))
+
+			// Verify chunk data packs are stored
+			for _, chunkDataPack := range chunkDataPacks {
+				_, err := chunkDataPackStore.ByChunkID(chunkDataPack.ChunkID)
+				require.NoError(t, err)
+			}
+
+			// Prepare chunk IDs for removal
+			chunkIDs := make([]flow.Identifier, len(chunkDataPacks))
+			for i, chunkDataPack := range chunkDataPacks {
+				chunkIDs[i] = chunkDataPack.ChunkID
+			}
+
+			// Test BatchRemove
+			require.NoError(t, protocolDB.WithReaderBatchWriter(func(protocolDBBatch storage.ReaderBatchWriter) error {
+				_, err := chunkDataPackStore.BatchRemove(chunkIDs, protocolDBBatch)
+				return err
+			}))
+
+			// Verify chunk data packs are removed from both protocol and chunk data pack DBs
+			for _, chunkID := range chunkIDs {
+				_, err := chunkDataPackStore.ByChunkID(chunkID)
+				assert.ErrorIs(t, err, storage.ErrNotFound)
+			}
+		})
 	})
 }
 
-// TestChunkDataPacks_MissingItem evaluates querying a missing item returns a storage.ErrNotFound error.
-func TestChunkDataPacks_MissingItem(t *testing.T) {
-	unittest.RunWithPebbleDB(t, func(pdb *pebble.DB) {
-		db := pebbleimpl.ToDB(pdb)
-		transactions := store.NewTransactions(&metrics.NoopCollector{}, db)
-		collections := store.NewCollections(db, transactions)
-		store1 := store.NewChunkDataPacks(&metrics.NoopCollector{}, db, collections, 1)
-
-		// attempt to get an invalid
-		_, err := store1.ByChunkID(unittest.IdentifierFixture())
-		assert.ErrorIs(t, err, storage.ErrNotFound)
+// TestChunkDataPacks_BatchRemoveChunkDataPacksOnly tests the BatchRemoveChunkDataPacksOnly method
+// which removes only from chunk data pack DB, leaving protocol DB mappings intact.
```

Comment:
inconsistent with implementation and test.

Reply:
updated